src/port1.0/portconfigure.tcl

```diff
@@ -528,98 +599,471 @@ proc portconfigure::configure_get_default_compiler {} {
+# https://gcc.gnu.org/wiki/openmp
+# https://trac.macports.org/wiki/XcodeVersionInfo
+#|---------------------------------------------------------------
+#| OpenMP Version |  Clang  |  Xcode Clang  |  Xcode  |   GCC   |
+#|---------------------------------------------------------------
+#|      2.5       |   3.8   |    Future?    | Future? |   4.2   |
+#|      3.0       |   3.8   |    Future?    | Future? |   4.4   |
+#|      3.1       |   3.8   |    Future?    | Future? |   4.7   |
+#|      4.0       | Partial |    Future?    | Future? |   4.9   |
+#|      4.5       | Partial |    Future?    | Future? |   ???   |
+#----------------------------------------------------------------
+#
+# utility procedure: get minimum command line compilers version based on restrictions
+proc portconfigure::get_min_command_line {compiler} {
+    global compiler.c_standard compiler.cxx_standard compiler.openmp_version compiler.thread_local_storage
+    set min_values 1.0
+    switch ${compiler} {
+        clang {
+            if {${compiler.c_standard} >= 2011} {
+                lappend min_values 318.0.61
+            } elseif {${compiler.c_standard} >= 1999} {
+                lappend min_values 211.10.1
             }
-            # No return here. 3.2.x with newer SDKs than 10.4u is handled below.
-        } elseif {[vercmp $xcodeversion 3.0] >= 0} {
-            if {"ppc" in $canonical_archs || "ppc64" in $canonical_archs} {
-                return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-gcc-6 macports-gcc-7}
-            } else {
-                return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-clang-3.4 macports-clang-3.3}
+            if {${compiler.cxx_standard} >= 2017} {
+                lappend min_values 902.0.39.1
+            } elseif {${compiler.cxx_standard} >= 2014} {
+                lappend min_values 600.0.54
+            } elseif {${compiler.cxx_standard} >= 2011} {
+                lappend min_values 500.2.75
             }
-        } else {
-            # Xcode 2.x (Tiger)
-            if {"ppc" in $canonical_archs || "ppc64" in $canonical_archs} {
-                if {"i386" in $canonical_archs} {
-                    # universal
-                    return {apple-gcc-4.2 gcc-4.0 macports-gcc-6 macports-gcc-7}
-                } else {
-                    # ppc only
-                    return {apple-gcc-4.2 gcc-4.0 gcc-3.3 macports-gcc-6 macports-gcc-7}
+            if {[option configure.cxx_stdlib] eq "libc++"} {
+                set cxx [lindex [split [portconfigure::configure_get_compiler cxx ${compiler}] /] end]
+                if {${cxx} ne "clang++"} {
+                    # 3.2 <= Xcode < 4.0 does not provide clang++
+                    lappend min_values [max_compiler_version]
                 }
+            }
+        }
+        llvm-gcc-4.2 -
+        gcc-4.2 -
+        gcc-4.0 -
+        apple-gcc-4.2 {
+            if {${compiler.c_standard} >= 1999} {
+                lappend min_values [max_compiler_version]
```

Comment:
You make a fair point, but I would argue that if we are going to have a `compiler.c_standard 1999`, then it should do what it advertises.

Reply:
Sure, I don't have a problem with having an option to request full C99 support, just with it being the default. That will require virtually everything to build with a macports-clang on older systems (and gcc-6 on powerpc), for example.

Reply:
Since there seems to be a consensus that `compiler.c_standard 1999` should indicated full C99 support, I will make this conversation as resolved.
We can continue the discussion about the **default** value of `compiler.c_standard` in the other conversation.